src/public/manageDomain.ts

```diff
@@ -0,0 +1,150 @@
+const envList = helper.getElement('.envList')
+const environmentVariables: EnvironmentItem[] = []
+const fullDomain = window.location.pathname.split('/').pop()
+document.getElementById('full-domain').innerText = fullDomain
+
+class EnvironmentItem {
+  name: string
+  value: string
+  isValid = false
+
+  constructor(name?: string, value?: string) {
+    this.name = name
+    this.value = value
+    const environmentElement = document.createElement('li')
+    environmentElement.classList.add(
+      'list-group-item',
+      'd-flex',
+      'align-items-center'
+    )
+    envList.appendChild(environmentElement)
+    environmentElement.innerHTML = `
+      <div class="input-group">
+        <input 
+          type="text"
+          aria-label="Name"
+          class="form-control text-uppercase text-monospace"
+          placeholder="NAME"
+          id="name"
+        >
+        <div class="input-group-prepend input-group-append">
+          <span class="input-group-text">=</span>
+        </div>
+        <input
+          type="text"
+          aria-label="Value"
+          class="form-control text-monospace"
+          placeholder="Value"
+          id="value"
+        >
+        <div class="input-group-append">
+          <button
+            class="btn btn-outline-danger"
+            type="button"
+            id="removeVariable"
+          >
+            Remove
+          </button>
+        </div>
+        <div class="invalid-feedback">
+          The variable name must contain only letters, number, and/or the underscore character.
+        </div>
+      </div>
+    `
+
+    const nameInputElement = environmentElement.querySelector(
+      '#name'
+    ) as HTMLInputElement
+    const valueInputElement = environmentElement.querySelector(
+      '#value'
+    ) as HTMLInputElement
+    nameInputElement.addEventListener('input', () => {
+      // validate the input for valid variable name with regex
+      // valid: letters, numbers and underscore
```

Comment:
```
const upperCaseValue = nameInputElement.value.toUpperCase()
const IS_INVALID = 'is-invalid'
const UNIQUE_REGEX_NAME = /^[A-Z0-9_]+$/g
```

Reply:
These are not completely necessary, but it makes it a bit DRYer and more consistent.
You can play with the variable names as well. Doesn't have to be exact.